packages/@react-stately/table/src/utils.ts

```diff
@@ -0,0 +1,116 @@
+import {GridNode} from '@react-types/grid';
+import {Key} from 'react';
+
+type mappedColumn<T> = GridNode<T> & {
+    index: number,
+    delta: number,
+    calculatedWidth?: number
+};
+
+export function mapToArray(columnWidthsMap: Map<Key, number>): { key: Key, width: number }[] {
+  return Array.from(columnWidthsMap, ([key, width]) => ({key, width}));
+}
+    
+export function getContentWidth(widths) {
+  return Array.from(widths).map(e => e[1]).reduce((acc, cur) => acc + cur, 0);
+}
+    
+export function isStatic(width: number | string): boolean {
+  return width !== null && width !== undefined && (typeof width === 'number' || width.match(/^(\d+)%$/) !== null);
+} 
+
+function parseFractionalUnit(width: string): number {
+  if (!width) {
+    return 1;
+  } 
+  let match = width.match(/^(\d+)(?=fr$)/);
+      // if width is the incorrect format, just deafult it to a 1fr
+  if (!match) {
+    console.warn(`width: ${width} is not a supported format, width should be a number (ex. 150), percentage (ex. '50%') or fr unit (ex. '2fr')`, 
+        'defaulting to \'1fr\'');
+    return 1;
+  }
+  return parseInt(match[0], 10);
+}
+
+export function parseStaticWidth(width: number | string, tableWidth: number): number {
+  if (typeof width === 'string') {
+    let match = width.match(/^(\d+)%$/);
+    if (!match) {
+      throw new Error('Only percentages or numbers are supported for static column widths');
+    }
+    return tableWidth * (parseInt(match[1], 10) / 100);
+  }
+  return width;
+}
+    
+    
+export function getMaxWidth(maxWidth: number | string, tableWidth: number): number {
+  return maxWidth !== undefined && maxWidth !== null
+        ? parseStaticWidth(maxWidth, tableWidth)
+        : Infinity;
+}
+
+export function getMinWidth(minWidth: number | string, tableWidth: number): number {
+  return minWidth !== undefined && minWidth !== null
+      ? parseStaticWidth(minWidth, tableWidth)
+      : 75;
+}
+
+function mapColumns<T>(remainingColumns: GridNode<T>[], remainingSpace: number, tableWidth: number): mappedColumn<T>[] {
```

Comment:
Isn't the PDF in this PR already? I'm not sure how I can add it as an artifact to the PR - I know it's in a comment but maybe there's a more "official" way to add it that you're referring to?

Reply:
Hmmm I don't see it in the Files changed list. I think we can just put it in that way though. It'd probably live in the RFC directory https://github.com/adobe/react-spectrum/tree/main/rfcs

Reply:
I see, so you mean you want it IN the codebase. I meant it was in a comment to the PR but I definitely never added it to the codebase.

I was actually thinking if we added it anywhere it would be in the docs but if you want it in the RFC directory I can put it there.